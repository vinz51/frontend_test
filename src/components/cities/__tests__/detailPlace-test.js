import CityMobileDetail, {
  CloseDetail,
  DetailPlace,
  Inhabitant,
} from "../detailPlace";
import City from "api/models/city";
import renderer from "react-test-renderer";
import React from "react";
import { shallow } from "enzyme";

describe("CloseDetail", () => {
  describe("Snapshots", () => {
    it("renders", () => {
      const myTreeComponent = renderer.create(<CloseDetail />).toJSON();

      expect(myTreeComponent).toMatchSnapshot();
    });
  });
});

describe("DetailPlace", () => {
  describe("Snapshots", () => {
    it("renders", () => {
      const myTreeComponent = renderer.create(<DetailPlace />).toJSON();

      expect(myTreeComponent).toMatchSnapshot();
    });
  });
});

describe("Inhabitant", () => {
  describe("Snapshots", () => {
    it("renders", () => {
      const myTreeComponent = renderer.create(<Inhabitant />).toJSON();

      expect(myTreeComponent).toMatchSnapshot();
    });
  });
});

describe("CityMobileDetail", () => {
  const myCity = new City({
    city: "Paris",
    state: "Ile de France",
    population: "2148000",
    growth_from_2000_to_2013: "3.33%",
  });

  describe("Snapshots", () => {
    it("renders", () => {
      const myTreeComponent = renderer
        .create(<CityMobileDetail city={myCity} />)
        .toJSON();

      expect(myTreeComponent).toMatchSnapshot();
    });
  });

  describe("Actions", () => {
    it("triggers onClose event", () => {
      const myOnClose = jest.fn();
      const myComponent = shallow(
        <CityMobileDetail city={myCity} onClose={myOnClose} />
      );

      expect(myOnClose).not.toHaveBeenCalled();

      myComponent.find(CloseDetail).simulate("click");

      expect(myOnClose).toHaveBeenNthCalledWith(1);

      myOnClose.mockClear();
    });
  });
});
