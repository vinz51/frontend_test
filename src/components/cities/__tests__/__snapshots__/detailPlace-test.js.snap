// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityMobileDetail Snapshots renders 1`] = `
<div
  className="sc-AxheI kElgGa"
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <div
    className="sc-AxhUy iFwpvP"
  >
    3.33%
  </div>
  <span
    className="sc-AxgMl hciPwI"
    onClick={null}
  >
    x
  </span>
  <h2
    className="sc-AxirZ sc-AxiKw hYWAVI"
  >
    Paris
  </h2>
  <h2
    className="sc-AxirZ sc-AxhCb dspALo"
  >
    State of 
    Ile de France
  </h2>
  <p
    className="sc-Axmtr HNdvW"
  >
    Inhabitant
    s
     
    <b>
      2148000
    </b>
  </p>
</div>
`;

exports[`CloseDetail Snapshots renders 1`] = `
<span
  className="sc-AxgMl hciPwI"
/>
`;

exports[`DetailPlace Snapshots renders 1`] = `
<div
  className="sc-AxheI kElgGa"
/>
`;

exports[`Inhabitant Snapshots renders 1`] = `
<p
  className="sc-Axmtr HNdvW"
/>
`;
